### PR TITLE
build(deps): bump k8s.io/client-go, k8s.io/apimachinery, k8s.io/api, …

### PR DESCRIPTION
**Pull request by jiuxia211**
…sigs.k8s.io/controller-runtime,github.com/onsi/gomega

### What this PR does / why we need it:

This PR updates multiple dependencies to their latest versions to ensure compatibility and benefit from the latest features and bug fixes. Specifically, it bumps the following dependencies:
- `k8s.io/client-go` from `0.26.3` to `0.30.2`
- `k8s.io/apimachinery` from `0.27.4` to `0.30.2`
- `k8s.io/api` from `0.26.3` to `0.30.2`
- `sigs.k8s.io/controller-runtime` from `0.14.6` to `0.18.4`
- `github.com/onsi/gomega  from` `1.30.0` to `1.33.1`

### Which issue(s) this PR fixes:

Fixes #

### Does this PR introduced a user-facing change?
None

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wenchajun** [comment]
Thanks for the PR. You'll need to sign your commit by:
```
git commit -s --amend
git push -f
```

**Comment by wenchajun** [comment]
I see 
```
./hack/update-codegen.sh
>> Temporary output directory /tmp/tmp.9aV294mKFK
ERROR: generate-groups.sh has been removed.
ERROR: Please use k8s.io/code-generator/kube_codegen.sh instead.
```

File: go.mod [hunk 1]
```diff
@@ -1,85 +1,85 @@
 module github.com/fluent/fluent-operator/v2
 
-go 1.21
+go 1.22.0
```
[the remark this answers is not in the document]
Comment (wenchajun) [reply go.mod:3]: The files in the dockerfile also need to be upgraded to the latest.

**Comment by jiuxia211** [comment]
> I see
> 
> ```
> ./hack/update-codegen.sh
> >> Temporary output directory /tmp/tmp.9aV294mKFK
> ERROR: generate-groups.sh has been removed.
> ERROR: Please use k8s.io/code-generator/kube_codegen.sh instead.
> ```

I updated update-codegen.sh following https://github.com/weaveworks/flagger/tree/main/hack but it doesn't seem to produce any output or changes and I'm not quite sure why this is happening

**Comment by benjaminhuo** [comment]
Great, thanks @jiuxia211 !
Looks like only one problem remains:
image

cc @wanjunlei

**Comment by wenchajun** [comment]
> > I see
> > ```
> > ./hack/update-codegen.sh
> > >> Temporary output directory /tmp/tmp.9aV294mKFK
> > ERROR: generate-groups.sh has been removed.
> > ERROR: Please use k8s.io/code-generator/kube_codegen.sh instead.
> > ```
> 
> I updated update-codegen.sh following https://github.com/weaveworks/flagger/tree/main/hack but it doesn't seem to produce any output or changes and I'm not quite sure why this is happening

You can run the command in your machine
```
make verify-codegen
```

```
>> Temporary output directory /tmp/tmp.oapWI9rc0l
ERROR: generate-groups.sh has been removed.
ERROR: Please use k8s.io/code-generator/kube_codegen.sh instead.

>> Removing /tmp/tmp.oapWI9rc0l
make: ***Makefile:91: verify-codegen] Error 1
```

**Comment by wenchajun** [comment]
https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L179
this shoud be updated too.

File: cmd/fluent-manager/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM --platform=$BUILDPLATFORM golang:1.21.5-alpine3.17 as builder
+FROM --platform=$BUILDPLATFORM golang:1.22.5-alpine3.20 as builder
```
Comment (wenchajun) [inline cmd/fluent-manager/Dockerfile:3]: Why is it 1.22.5 and not 1.22.0?

File: cmd/fluent-watcher/fluentd/Dockerfile.amd64 [hunk 3]
```diff
@@ -8,54 +10,25 @@ RUN echo $(ls -al /code)
...
  && gem install fluentd -v 1.17.0 \
- && gem install bigdecimal -v 1.4.4 \
-# NOTE: resolv v0.2.1 includes the fix for CPU spike issue due to DNS resolver.
-# This hack is needed for Ruby 2.6.7, 2.7.3 and 3.0.1. (alpine image is still kept on 2.7.3)
- && gem install resolv -v 0.2.1 \
- && gem install elasticsearch -v 7.13.3 \
- && gem install elasticsearch-xpack -v 7.13.3 \
- && gem install fluent-plugin-detect-exceptions -v 0.0.14 \
- && gem install \
-         fluent-plugin-s3 \
```
Comment (benjaminhuo) [inline cmd/fluent-watcher/fluentd/Dockerfile.amd64:39]: the built-in fluentd plugins shouldn't be remove I think

Comment (benjaminhuo) [reply cmd/fluent-watcher/fluentd/Dockerfile.amd64:39]: @wanjunlei @wenchajun what do you think?

File: cmd/fluent-manager/Dockerfile (hunk 2 above)
Comment (benjaminhuo) [reply cmd/fluent-manager/Dockerfile:3]: Go released 1.22.5 recently, it might be ok?

File: cmd/fluent-watcher/fluentd/Dockerfile.amd64 (hunk 3 above)
Comment (benjaminhuo) [reply cmd/fluent-watcher/fluentd/Dockerfile.amd64:39]: in the latest fluentd build workflow:
- fluentd image is actually first build and pushed to ghcr, not dockerhub https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L252-L268
https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L294-L310
https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L471-L472
- And after all the arm64 and amd64 fluentd images are built and push to ghcr, all these images are synced to dockerhub kubespere org https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L494-L505

Using a image on dockerhub kubesphere org as base image is not ok in my opinion

Comment (jiuxia211) [reply cmd/fluent-watcher/fluentd/Dockerfile.amd64:39]: Sorry, this is a modification I didn't realize. I made a mistake when I copied it.

File: cmd/fluent-manager/Dockerfile (hunk 2 above)
Comment (jiuxia211) [reply cmd/fluent-manager/Dockerfile:3]: I wiil change it

**Review by wenchajun — APPROVED** [review]

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
This is great, thank you @jiuxia211 !

**Comment by benjaminhuo** [comment]
@jiuxia211 is it possible for you to upgrade these new dependencies?

Thanks 

image

**Comment by jiuxia211** [comment]
> @jiuxia211 is it possible for you to upgrade these new dependencies?
> 
> Thanks
> 
> image

OK, assign it to me

**Pull request merged**